torch/testing/_internal/common_state_dict.py

```diff
@@ -0,0 +1,110 @@
+# Owner(s): ["oncall: distributed"]
+
+import copy
+from itertools import chain
+from typing import Any, Dict
+
+import torch
+import torch.nn as nn
+from torch.distributed._shard.sharded_tensor import ShardedTensor
```

Comment:
Looks like we are hitting a circular dependency. 
https://hud.pytorch.org/pr/pytorch/pytorch/113181#18462382108

(https://www.internalfb.com/code/fbsource/[aa844ff7814ae9b8ac85e97059fda866ac975a72]/fbcode/caffe2/torch/distributed/_shard/api.py?lines=5) We might be able to get around it by dropping the import at the top and directly import the function being used in c10d.

Reply:
Since this is a test util, maybe we can just perform lazy initialization to avoid the circular dependency.

Reply:
Thanks for catching, was a little confused by the failing test!